### ISSUE TITLE
Double-encoded JSON fails

### ISSUE THREAD

**Issue by rexxars** [queckezz/json-tokenize#4]
Basically, if you've got a blob of JSON inside of another JSON string, the tokenizing fails:

```
  /home/espenh/webdev/json-tokenize/dist.js:65

   64:   }, null);                         
   65:   var createToken = ref.createToken;
   66:   var str = ref.str;                

  Error: Cannot read property 'createToken' of null

  tokenize (dist.js:65:24)
  next (dist.js:95:12)
  tokenize (dist.js:85:10)
  next (dist.js:95:12)
  tokenize (dist.js:85:10)
  next (dist.js:95:12)
  tokenize (dist.js:85:10)
  next (dist.js:95:12)
  tokenize (dist.js:85:10)
  next (dist.js:95:12)
  tokenize (dist.js:85:10)
  next (dist.js:95:12)
  tokenize (dist.js:85:10)
  next (dist.js:95:12)
  tokenize (dist.js:85:10)
  next (dist.js:95:12)
  tokenize (dist.js:85:10)
```

Test case:

```js
test('double-encoded json', (t) => {
  const inner = JSON.stringify({ x: 1, y: '2' })
  const outer = JSON.stringify({ type: 'blob', data: inner })
  const innerEnc = JSON.stringify(inner)
  validateJson(t, outer, [
    { type: 'punctuation', raw: '{', value: '{' },
    { type: 'string', raw: '"type"', value: 'type' },
    { type: 'punctuation', raw: ':', value: ':' },
    { type: 'string', raw: '"blob"', value: 'blob' },
    { type: 'punctuation', raw: ',', value: ',' },
    { type: 'string', raw: '"data"', value: 'data' },
    { type: 'punctuation', raw: ':', value: ':' },
    { type: 'string', raw: innerEnc, value: inner },
    { type: 'punctuation', raw: '}', value: '}' }
  ])
})
```

**Comment by queckezz** [comment]
Thanks for the test case! I just pushed a fix for this, released as `0.3.0` :sparkles:

**Comment by rexxars** [comment]
Thanks!
